src/streaming/protection/controllers/ProtectionController.js

```diff
@@ -384,6 +384,27 @@ function ProtectionController(config) {
             return;
         }
 
+        // Enforce maximum number of open MediaKeySessions, only if keepProtectionMediaKeys is true
+        if (settings) {
```

Comment:
Can we add this logic to a separate private function like `_enforceMediaKeySessionLimit()`